### PR TITLE
expression: support for testing a single vectorized evaluation function

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve? 
Support for testing a single vectorized evaluation function.

### What is changed and how it works?
Use `go test -args` to decide the test which function signature you want to run.

For example, if you want to test the `builtinLog10Sig`  and `builtinSqrtSig`. You can input ```go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMathFunc -run=BenchmarkVectorizedBuiltinMathFunc -args "builtinLog10Sig" "builtinSqrtSig"``` .  Just input the function signature behind the "-args" flag. If you want to run all of the test, just ignore the "-args" flag.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -511,13 +512,27 @@ func genVecBuiltinFuncBenchCase(ctx sessionctx.Context, funcName string, testCas
 	return baseFunc, input, result
 }
 
-// testVectorizedBuiltinFunc is used to verify that the special vectorized
+// testVectorizedBuiltinFunc is used to verify that the vectorized
 // expression is evaluated correctly
 func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
+	// If you just want to run a special vectorized expression evaluation.
```
Comment (qw4990) [inline expression/bench_test.go]:
```suggestion
	// If you just want to run a specified vectorized expression.
```

File: expression/bench_test.go [hunk 2]
```diff
@@ -511,13 +512,27 @@ func genVecBuiltinFuncBenchCase(ctx sessionctx.Context, funcName string, testCas
...
-// testVectorizedBuiltinFunc is used to verify that the special vectorized
+// testVectorizedBuiltinFunc is used to verify that the vectorized
 // expression is evaluated correctly
 func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
+	// If you just want to run a specified vectorized built-in function.
+	// You should input "export vec_test_func_sig=XXXsig" first, which "XXXsig" means
+	// the expression function signature you want to test.
+	// If you want to test all of them, just ignore it or "export vec_test_func_sig="
+	// The following benchmark is also like this.
+	TestFuncName := os.Getenv("vec_test_func_sig")
```
Comment (lamxTyler) [inline expression/bench_test.go]: Just wondering can we use golang flag? It can be passed with `go test`.

Comment (Reminiscent) [reply expression/bench_test.go]: I think your opinion is better, I will do some investigation

Comment (Reminiscent) [reply expression/bench_test.go]: update. PTAL

File: expression/bench_test.go [hunk 3]
```diff
@@ -520,13 +521,30 @@ func genVecBuiltinFuncBenchCase(ctx sessionctx.Context, funcName string, testCas
...
 
-// testVectorizedBuiltinFunc is used to verify that the special vectorized
+// testVectorizedBuiltinFunc is used to verify that the vectorized
 // expression is evaluated correctly
 func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
+	flag.Parse()
+	var testFunc map[string]bool
+	testFunc = make(map[string]bool)
+	argList := flag.Args()
+	testAll := true
```
Comment (zz-jason) [inline expression/bench_test.go]: how about:
```go
testAll := len(argList) > 0
```

Comment (Reminiscent) [reply expression/bench_test.go]: update. PTAL

File: expression/bench_test.go [hunk 4]
```diff
@@ -520,13 +521,29 @@ func genVecBuiltinFuncBenchCase(ctx sessionctx.Context, funcName string, testCas
 	return baseFunc, input, result
 }
 
-// testVectorizedBuiltinFunc is used to verify that the special vectorized
+// testVectorizedBuiltinFunc is used to verify that the vectorized
 // expression is evaluated correctly
 func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
+	flag.Parse()
```
Comment (lamxTyler) [inline expression/bench_test.go]: It is not necessary to call it here, see https://github.com/golang/go/issues/31859.

File: expression/bench_test.go [hunk 5]
```diff
@@ -645,16 +662,29 @@ func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
 }
 
 // benchmarkVectorizedBuiltinFunc is used to get the effect of
-// using the special vectorized expression evaluations
+// using the vectorized expression evaluations
 func benchmarkVectorizedBuiltinFunc(b *testing.B, vecExprCases vecExprBenchCases) {
 	ctx := mock.NewContext()
+
+	flag.Parse()
```
Comment (lamxTyler) [inline expression/bench_test.go]: Ditto.

File: expression/bench_test.go [hunk 6]
```diff
@@ -645,16 +662,29 @@ func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
...
+	var testFunc map[string]bool
+	testFunc = make(map[string]bool)
```
Comment (lamxTyler) [inline expression/bench_test.go]: Ditto.

**Comment by Reminiscent** [comment]
@lamxTyler  Update. PTAL

**Pull request merged**